internal/npm_install/generate_build_file.ts

```diff
@@ -171,10 +171,23 @@ for installation instructions.`);
  */
 function generateRootBuildFile(pkgs: Dep[]) {
   let exportsStarlark = '';
-  pkgs.forEach(pkg => {pkg._files.forEach(f => {
-                 exportsStarlark += `    "node_modules/${pkg._dir}/${f}",
-`;
-               })});
+  for (const pkg of pkgs) {
+    // export the package dir itself so that it can be reference
+    // by the `@wksp//:node_modules/pkg` label
+    exportsStarlark += `    "node_modules/${pkg._dir}",\n`;
+    const dirs = [];
+    for (const f of pkg._files) {
+      dirs.push(path.dirname(f))
+      exportsStarlark += `    "node_modules/${pkg._dir}/${f}",\n`;
+    }
+    // export the all dirs so that they can be reference
+    // by the `@wksp//:node_modules/pkg/path/to/dir` label
+    for (const d of [...new Set(dirs)]) {
```

Comment:
I'm a little worried about the generated file getting "too big" whatever that means.

Do you actually use the subdirectories also? isn't it enough to get the top-level packages?

Reply:
Its already pretty big :) I only need the top-level package for angular. Besides the size I didn't see any reason to not include the nested directories incase someone wants to reference them as labels. Maybe that reason is enough to not include them however. Globbed exports_files would be nice.